drivers/accel/adxl313/adxl313.c

```diff
@@ -488,29 +488,184 @@ int adxl313_get_op_mode(struct adxl313_dev *dev,
 *******************************************************************************/
 int adxl313_set_offset(struct adxl313_dev *dev, int32_t offset_ug,
 		       enum adxl313_axis axis)
+{
+	int32_t offset_val = 0x00;
+
+	if (axis > ADXL313_Z_AXIS)
+		return -EINVAL;
+
+	switch (dev->dev_type) {
+	case ID_ADXL312:
+		offset_val = offset_ug / ADXL312_OFFSET_SCALE_FACTOR;
+		break;
+	case ID_ADXL313:
+		offset_val = offset_ug / ADXL313_OFFSET_SCALE_FACTOR;
+		break;
+	case ID_ADXL314:
+		offset_val =  offset_ug / ADXL314_OFFSET_SCALE_FACTOR;
+		break;
+	default:
+		return -ENODEV;
+	}
+
+	switch (dev->dev_type) {
+	case ID_ADXL312:
+		if (dev->resolution == ADXL313_10_BIT_RES)
+			offset_val = (offset_val * 4) >> (dev->range - ADXL313_RANGE_FACTOR);
+		else
+			offset_val = offset_val * 4;
+		break;
+	case ID_ADXL313:
+		if (dev->resolution == ADXL313_10_BIT_RES) {
+			offset_val = (offset_val * 4);
+			offset_val = offset_val >> dev->range;
+		} else
+			offset_val = offset_val * 4;
+		break;
+	case ID_ADXL314:
+		offset_val = offset_val * 4;
+		break;
+	default:
+		return -ENODEV;
+	}
+
+	/* Write raw data to register for setting offset. */
+	return adxl313_set_raw_offset(dev, offset_val, axis);
+}
+
+/*******************************************************************************
+ * @brief Get offset for each axis.
+ *
+ * @param dev       - The device structure.
+ * @param offset_ug - Offset read in g / 1000 000 (ug).
+ * @param axis      - Axis to apply offset.
+ *
+ * @return 0 in case of success, negative error code otherwise.
+*******************************************************************************/
+int adxl313_get_offset(struct adxl313_dev *dev, int32_t *offset_ug,
+		       enum adxl313_axis axis)
 {
 	int ret;
-	uint8_t reg_val = 0x00;
+	int32_t raw_offset = 0x00;
+
+	if (axis > ADXL313_Z_AXIS)
+		return -EINVAL;
+
+	/* Read data to register for getting offset register value. */
+	ret = adxl313_get_raw_offset(dev, &raw_offset, axis);
+	if (!ret) {
```

Comment:
please use `if (ret)` and `return ret;` here.

Reply:
Thank you for the observation! I updated the code.